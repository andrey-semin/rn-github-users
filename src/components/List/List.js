import React from 'react'
import PropTypes from 'prop-types'
import { FlatList } from 'react-native'

import ListSeparator from './ListSeparator'

import styles from './styles'

const List = ({ data, renderItem }) => (
  <FlatList
    style={styles.root}
    data={data}
    renderItem={renderItem}
    ItemSeparatorComponent={ListSeparator}
  />
)

List.propTypes = {
  data: PropTypes.array.isRequired,
  renderItem: PropTypes.func.isRequired
}

export default List
