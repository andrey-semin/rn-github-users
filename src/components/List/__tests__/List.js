import React from 'react'
import { Text } from 'react-native'
import renderer from 'react-test-renderer'

import List from '../List'

it('List renders correctly', () => {
  const tree = renderer
    .create(
      <List
        data={[{ id: 1 }, { id: 2 }]}
        renderItem={({ item }) => <Text>{item.id}</Text>}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
