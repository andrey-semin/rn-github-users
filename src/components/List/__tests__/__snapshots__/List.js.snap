// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  data={
    Array [
      Object {
        "id": 1,
      },
      Object {
        "id": 2,
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onViewableItemsChanged={undefined}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "backgroundColor": "#fefefe",
    }
  }
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        1
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#f2f2f2",
            "flex": 1,
            "height": 1,
          }
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        2
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
